tests/units/units.cpp

```diff
@@ -791,6 +791,12 @@ TEST(Units, unitAttributeReference)
 
     EXPECT_EQ("NewUnit", u->unitAttributeReference(0));
     EXPECT_EQ("", u->unitAttributeReference(4));
+
+    u->setUnitAttributeReference(0, "NewerUnit");
+    EXPECT_EQ("NewerUnit", u->unitAttributeReference(0));
+
+    u->setUnitAttributeReference(3, "FreshUnit");
+    EXPECT_EQ("NewerUnit", u->unitAttributeReference(0));
```

Comment:
Is this a useful test? is it just checking that the new method is not doing something it shouldn't do?

Reply:
Probably not.

Reply:
[61fa17c](https://github.com/cellml/libcellml/pull/1167/commits/61fa17c3e8c2111e15c9f43a6a6434890504f428)

Reply:
It is a useful test, it is checking what happens when we try to set an out-of-bounds attribute value.